WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserFragment.kt

```diff
@@ -0,0 +1,433 @@
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        @Suppress("SwallowedException")
+        callback = try {
+            activity as ThemeBrowserFragmentCallback
+        } catch (e: ClassCastException) {
+            throw ClassCastException("$activity must implement ThemeBrowserFragmentCallback")
+        }
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        searchView?.setOnQueryTextListener(null)
+        callback = null
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = ThemeBrowserFragmentBinding.inflate(inflater, container, false)
+
+        configureGridView(inflater)
+        configureSwipeToRefresh()
+
+        return binding.root
+    }
+
+    override fun onResume() {
+        super.onResume()
+        (activity as? ScrollableViewInitializedListener)?.onScrollableViewInitialized(binding.themeListview.id)
+    }
+
+    override fun onDestroyView() {
+        _binding = null
+        super.onDestroyView()
+    }
+
+    @Suppress("deprecation")
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        adapter.setThemeList(fetchThemes())
+        binding.themeListview.adapter = adapter
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        if (searchMenuItem != null && requireNotNull(searchMenuItem).isActionViewExpanded) {
+            outState.putString(KEY_LAST_SEARCH, searchView?.query.toString())
+        }
+        outState.putParcelable(QuickStartEvent.KEY, quickStartEvent)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.search, menu)
+        searchMenuItem = menu.findItem(R.id.menu_search)
+        searchView = searchMenuItem?.actionView as SearchView
+        searchView?.setOnQueryTextListener(this)
+        searchView?.maxWidth = Int.MAX_VALUE
+
+        if (!TextUtils.isEmpty(lastSearch)) {
+            searchMenuItem?.expandActionView()
+            onQueryTextSubmit(lastSearch)
+            searchView?.setQuery(lastSearch, true)
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == R.id.menu_search) {
+            AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.THEMES_ACCESSED_SEARCH, site)
+            return true
+        }
+        @Suppress("DEPRECATION")
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onQueryTextSubmit(query: String?): Boolean {
+        adapter.filter.filter(query)
+        searchView?.clearFocus()
+        return true
+    }
+
+    override fun onQueryTextChange(newText: String?): Boolean {
+        adapter.filter.filter(newText)
+        return true
+    }
+
+    override fun onMovedToScrapHeap(view: View?) {
```

Comment:
Since this view is coming as nullable from [RecyclerListener.onMovedToScrapHeap()](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/AbsListView.java#7093), and there are no nullability annotations in the Android SDK's Java file, I kept it as nullable when overriding it in the `ThemeBrowserFragment`.

Reply:
> Since this view is coming as nullable from [RecyclerListener.onMovedToScrapHeap()](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/AbsListView.java#7093)

What made you think it is coming as nullable @irfano ? To me it looks like the nullability is unknown, just like you suggested, because there are no nullability annotations defined there. But, to be honest, I just want to understand more on how you derived to that conclusion, that this is coming as nullable from the [docs](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/AbsListView.java#7093) you shared. It will help me understand on how to do that better next time as well. 🙏 

FYI: I have just made an assumption that this is never going to be nullable as there is no point in calling the `onMovedToScrapHeap(...)` from a nullable view, for example, I applied the same logic on `onClick(...)`, which has the same problem of unknown nullability... 🤷

Reply:
I think we perceive nullability from Java classes differently, and it's good to understand our perspectives.
I see everything from the Java file as nullable if non-null is not explicitly stated. Thinking about whether the `view` from `onMovedToScrapHeap(View view)` can be null or not shouldn't be the responsibility of the library consumer (me). I cannot assume that it won't be null. It's possible that it might be null when the app is in the background with low memory, but the `RecyclerListener` is still active 🤷🏻‍♂️. I think keeping these nullable is more effective than thinking about whether functions of libraries can really be null or not.

Reply:
Totally, this is a very correct perception, and someone could argue that this discussion we are having is about defensive vs offensive programming. For few years now, I am a proponent of offensive programming, and thus, this explains why I am thinking and acting this way. IMHO it is more valuable to risk a 0.1% crash than support dead code as usually having something as "fake" nullable has this effect. Still, being only 99.9% right has it risks and I value that too.

Let's resolve this for now, but, no matter, this was such a good discussion and it is great for me to know where you stand on that and how you think about it @irfano , thanks many! 💯 🙇 🌟